pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1870,8 +1884,14 @@ public static WorkerConfig initializeWorkerConfigFromBrokerConfig(ServiceConfigu
         workerConfig.setTlsEnableHostnameVerification(brokerConfig.isTlsHostnameVerificationEnabled());
         workerConfig.setBrokerClientTrustCertsFilePath(brokerConfig.getBrokerClientTrustCertsFilePath());
         workerConfig.setTlsTrustCertsFilePath(brokerConfig.getTlsTrustCertsFilePath());
+        workerConfig.setTlsKeyFilePath(brokerConfig.getTlsKeyFilePath());
+        workerConfig.setTlsCertificateFilePath(brokerConfig.getTlsCertificateFilePath());
 
         // client in worker will use this config to authenticate with broker
+        if (isNotBlank(brokerConfig.getBrokerClientAuthenticationPlugin())
+                && isNotBlank(brokerConfig.getBrokerClientAuthenticationParameters())) {
+            workerConfig.setBrokerClientAuthenticationEnabled(true);
```

Comment:
Same as above, we should respect the user configured `brokerClientAuthenticationEnabled`.